lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -51,6 +54,50 @@ export class ResponseHandler {
             tenantId: idTokenObj.claims.tid,
         };
     }
+
+    private saveToken(originalTokenResponse: TokenResponse, authority: string, serverTokenResponse: ServerAuthorizationTokenResponse, clientInfo: ClientInfo): TokenResponse {
+        const tokenResponse = { ...originalTokenResponse };
+        // Set consented scopes in response
+        const requestScopes = ScopeSet.fromString(serverTokenResponse.scope, this.clientId, false);
+        tokenResponse.scopes = requestScopes.asArray();
+
+        const accessTokenCacheItems = this.cacheManager.getAllAccessTokens(this.clientId, authority);
+
+        for (let i = 0; i < accessTokenCacheItems.length; i++) {
+            const accessTokenCacheItem = accessTokenCacheItems[i];
+            if (accessTokenCacheItem.key.homeAccountIdentifier === tokenResponse.account.homeAccountIdentifier) {
+                const cachedScopes = ScopeSet.fromString(accessTokenCacheItem.key.scopes, this.clientId, false);
+
+                if(cachedScopes.intersectingScopeSets(tokenResponse.scopes)) {
+                    this.cacheStorage.removeItem(JSON.stringify(accessTokenCacheItem.key));
+                }
+            }
+        }
+
+        // Expiration calculation
+        const expiresIn = TimeUtils.parseExpiresIn(serverTokenResponse.expires_in);
+        const expiration = TimeUtils.now() + expiresIn;
```

Comment:
Can we label these as `seconds` to make it clear which units are used?

Reply:
+1

Reply:
+1

Reply:
This is done in a later PR.